### PR TITLE
Store work identifiers in catalogue graph

### PR DESCRIPTION
**Pull request by StepanBrychta**
## What does this change?

https://github.com/wellcomecollection/platform/issues/6087

Add a new catalogue graph extractor to:
* Create separate `PathIdentifier` nodes from collection path fragments associated with works
* Create parent/child (`HAS_PARENT`) edges between path identifiers
* Create `HAS_PATH_IDENTIFIER` edges between work nodes and identifier nodes

Capturing this information in the catalogue graph is the first step towards replacing the relation embedder subsystem.

This implementation slightly deviates from RFC 076, which suggested to make separate nodes for **all** identifiers associated with a given work. Making identifier nodes only based on collection paths is simpler and more performant, and handles edge cases where a collection path of a given work does not match any of the work's identifiers.

## How to test

Run the new `catalogue_work_identifiers` extractor locally for both `nodes` and `edges`. Check the output.

## How can we measure success?

Once added into the catalogue graph, the new nodes and edges should allow us to construct work hierarchies at query time, reproducing the functionality of the relation embedder.

## Have we considered potential risks?

There are no immediate risks — this PR does not make changes to existing nodes/edges used to populate the production concepts index.  

Longer-term risks which come with replacing the current relation embedder subsystem include:
* Failing to create `HAS_PARENT` edges between path identifiers which should be linked
* Incorrectly creating `HAS_PARENT` edges between path identifiers which should **not** be linked

To verify that the implementation avoids these two issues, I compared the parent/child relationships created by the new `catalogue_work_identifiers` extractor with those created by the relation embedder. (Relationships created by the relation embedder were retrieved from the denormalised index using the `ancestors` field. Relationships created by the extractor were retrieved from the graph via a Cypher query.)

The relation embedder created 261,470 parent/child relationships, while the extractor created 261,720. All of the relationships created by the relation embedder were correctly reproduced in the graph. However, the extractor created 250 additional relationships.

Below is a CSV file storing a full list of these new relationships. Having checked several of the child works, I think that these relationships **should** exist, since all of the child works I checked have siblings for which the same parent relationship does exist. I'm not sure why these relationships were not created by the relation embedder.

[relationship_diff.csv](https://github.com/user-attachments/files/21412769/relationship_diff.csv)

### CODE REVIEW

File: catalogue_graph/tests/fixtures/catalogue/denormalised_works_example.jsonl [hunk 1]
```diff
@@ -1,3 +1,3 @@
...
...entified"}, "label": "Etching", "type": "GenreConcept"}]}], "contributors": [{"id": {"type": "Unidentifiable"}, "agent": {"id": {"canonicalId": "uykuavkt", "sourceIdentifier": {"identifierType": {"id": "label-derived"}, "ontologyType": "Person", "value": "gamelin, jacques, 1739-1803"}, "otherIdentifiers": [], "type": "Identified"}, "label": "Gamelin, Jacques, 1739-1803.", "type": "Person"}, "roles": [], "primary": true}, {"id": {"type": "Unidentifiable"}, "agent": {"id": {"canonicalId": "h2peyydc", "sourceIdentifier": {"identifierType": {"id": "lc-names"}, "ontologyType": "Person", "value": "nr90012131"}, "otherIdentifiers": [], "type": "Identified"}, "label": "Mart\u00edn.", "type": "Person"}, "roles": [], "primary": false}], "workType": "Standard", "otherIdentifiers":  [{"value": "1569755", "identifierType":  {"id": "sierra-identifier"}}, {"value": "569755i", "identifierType":  {"id": "iconographic-number"}}, {"value": "V0008815", "identifierType":  {"id": "miro-image-number"}}], "collectionPath":  {"path": "147150i/569755i" }}, "state": {"canonicalId": "m4u8drnu", "sourceIdentifier":  {"value": "b15697551", "identifierType":  {"id": "sierra-system-number"}}}, ",type": "Visible"}
```
[the remark this answers is not in the document]
Comment (kenoir) [reply catalogue_graph/tests/fixtures/catalogue/denormalised_works_example.jsonl:1]: This isn't correct.

**Review by kenoir — APPROVED** [review]
I needed to pin the ES version to get this to work, and added some code to allow early termination of sources (to prevent the local hanging issue when queues are not exhausted when a sample size is specified).
